src/ui/pages/Security/ChangePassword.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import { getError } from 'src/shared/errors/getError';
+import { invariant } from 'src/shared/invariant';
+import { useBackgroundKind } from 'src/ui/components/Background';
+import { whiteBackgroundKind } from 'src/ui/components/Background/Background';
+import { PageColumn } from 'src/ui/components/PageColumn';
+import { PageHeading } from 'src/ui/components/PageHeading';
+import { PageTop } from 'src/ui/components/PageTop';
+import { ViewSuspense } from 'src/ui/components/ViewSuspense';
+import { accountPublicRPCPort } from 'src/ui/shared/channels';
+import { zeroizeAfterSubmission } from 'src/ui/shared/zeroize-submission';
+import { Button } from 'src/ui/ui-kit/Button';
+import { Input } from 'src/ui/ui-kit/Input';
+import { UIText } from 'src/ui/ui-kit/UIText';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import { ChangePasswordSuccess } from './ChangePasswordSuccess';
+
+function CustomValidityInput({
+  customValidity,
+  ...props
+}: React.InputHTMLAttributes<HTMLInputElement> & { customValidity: string }) {
+  const ref = useRef<HTMLInputElement>(null);
+  useLayoutEffect(() => {
```

Comment:
Can we just use `useCustomValidity` here instead of new component?

Reply:
We can, but the point is to test the abstraction at a component level :)